book/src/qobject/qobject_struct.md

```diff
@@ -5,42 +5,134 @@ SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+
+[Full Example](https://github.com/KDAB/cxx-qt/blob/main/examples/qml_features/rust/src/properties.rs)
+
+### `cxx_type`
+
+You can change the C++ type that your property uses by adding the `cxx_type` attribute to the `#[qproperty]` macro.
+
+This is especially useful if your property type is an opaque C++ type and can't be owned directly by Rust.
+Then it might be necessary to wrap it in a [cxx::UniquePtr](https://docs.rs/cxx/latest/cxx/struct.UniquePtr.html).
+
+However, you may still want your `Q_PROPERTY` to use the opaque type directly.
+This can easily be achieved by using `cxx_type`.
+
+``` rust,noplayground
+#[cxx_qt::qobject]
+struct MyStruct {
+  #[qproperty(cxx_type="Opaque")]
+  my_property: cxx::UniquePtr<Opaque>
+}
 ```
+In this case, CXX-Qt will automatically convert any references to the `std::unique_ptr<Opaque>` to references to `Opaque`.
+
+For details see the [page on type conversions](../concepts/type-conversions.md).
 
 ## Invokables
 
-A `impl qobject::T` is used to define invokables, the `impl qobject::T` defines that the methods are implemented onto the C++ QObject `T`.
-Therefore they have access to both C++ and Rust methods. Also CXX-Qt adds wrapper code around your invokables to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
+Invokables are functions that are registered by with the Qt meta-object system using [`Q_INVOKABLE`](https://doc.qt.io/qt-6/qobject.html#Q_INVOKABLE).
+In practice this most importantly means they can be called from QML, as well as C++.
 
-To mark a method as invokable simply add the `#[qinvokable]` attribute to the Rust method. This then causes `Q_INVOKABLE` to be set on the C++ definition of the method, allowing QML to call the invokable.
+CXX-Qt allows you to define invokables using Rust code.
+This way you can easily add a Rust-powered backend to your QML frontend.
 
-Note to access properties on the C++ object use [Cpp Object](./cpp_object.md).
+Invokables, by definition, must be defined on a C++ class however.
+That's where the QObject subclass generated by `#[cxx_qt::qobject]` comes into play.
+For details on this, see the [`qobject::T` page](./cpp_object.md).
 
-## Properties
+The important part for invokables is that they need to be implemented on the `qobject::T`, not `T`.
+Therefore they have access to both C++ and Rust methods. Also CXX-Qt adds wrapper code around your invokables to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
 
-Fields within the `#[cxx_qt::qobject]` marked struct can be tagged with the `#[qproperty]`, this declares which fields should be exposed as `Q_PROPERTY`s on the QObject.
+To mark a method as invokable simply add the `#[qinvokable]` attribute to the Rust method. This then causes the method to be exposed on the C++ class.
+`Q_INVOKABLE` will also be added to the definition of the method, allowing QML to call the invokable.
```

Comment:
This is a bit unclear. Is it not exposed to C++ (without `Q_INVOKABLE`) if `#[qinvokable]` isn't used?

Reply:
Oh, you answer this below :)